src/tool/cppwinrt/test/GetMany.cpp

```diff
@@ -15,33 +15,51 @@ using namespace Windows::Foundation::Collections;
 
 namespace
 {
-    template <typename T>
-    IIterable<T> single_threaded_list(std::list<T>&& values = {})
+    template<typename T>
+    IIterator<T> single_threaded_generator(std::vector<T>&& values = {})
     {
-        struct list :
-            implements<list, IIterable<T>>,
-            iterable_base<list, T>
+        // This iterator may only be advanced once, ensuring the GetMany complexity optimization
+        // is actually enforced with this test.
+
+        struct generator_container
         {
-            explicit list(std::list<T>&& values) : m_values(std::forward<std::list<T>>(values))
+            explicit generator_container(IIterator<T> const& first) : m_current(first)
+            {
+                if (!m_current.HasCurrent())
+                {
+                    m_current = nullptr;
+                }
+            }
+
+            IIterator<T> begin() const { return m_current; }
+            IIterator<T> end() const { return nullptr; }
+
+        private:
+            IIterator<T> m_current;
+        };
+
+        struct generator : implements<generator, IIterable<T>>, iterable_base<generator, T>
+        {
+            explicit generator(IIterator<T> const& first) : m_container(first)
             {
             }
 
             auto& get_container() noexcept
             {
-                return m_values;
+                return m_container;
             }
 
             auto& get_container() const noexcept
             {
-                return m_values;
+                return m_container;
             }
 
         private:
-
-            std::list<T> m_values;
+            generator_container m_container;
         };
 
-        return make<list>(std::move(values));
+        auto v = single_threaded_vector<T>(std::move(values));
```

Comment:
Add the ability to do this for a container that is definitely not random-access, like `std::list`.

Reply:
That's what `generator_container` ensures as it only offers an `IIterator<T>`. I initially used `std::list` alone and while that ensures that random access is not available, it doesn't ensure that multiple traversals aren't employed. Here a vector is used merely for convenience to create the container that the iterator is backed by.